### ISSUE TITLE
How to Split up/Unstack/Partition a dynamic 3D Tensor into subtensors?

### ISSUE THREAD

**Issue by Aashit-Sharma** [tensorflow/tensorflow#17317]
In this [code from google][1] , sequence loss is being calculated by passing in 3 variables : logits , weights and targets.

How logits are defined:
> logits: A Tensor of shape
>       `[batch_size, sequence_length, num_decoder_symbols]` and dtype float.
>       The logits correspond to the prediction across all classes at each
>       timestep.

My intentions were to get **3 different tensors of shapes** batch_size , sequence_length and dec_symbols  and then use them in **tf.scan** (using sequence_length as elems)

If I print the logits tensor , this is what i get : 
> shape=(?, ?, 300) 

Which means tf.unstack is out of the equation(as it has a variable shape)

So my first question is , is it even possible ?

If yes , any suggestions ?

Thanks !

PS: Maybe Google can add a swap_memory parameter to the seq2seq.sequence_loss function ,to avoid OOM errors while calculating losses [which is what we are trying to overcome using an iterator inside]

  [1]: https://github.com/tensorflow/tensorflow/blob/r1.5/tensorflow/contrib/seq2seq/python/ops/loss.py

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this still an issue?
